### DISCUSSION TITLE
Clear the remote cache?

### DISCUSSION THREAD

**Discussion by matthias-ccri** [nrwl/nx#28565] [Feature Requests]
Right now at my org we're seeing very intermittent build failures that are caused by invalid cache entries. We use a remote cache with Minio. We don't yet know what is causing these invalid cache entries to be populated, but it would be nice to be able to clear them.

Is there a way within the nx toolchain to clear cache entries in a remote cache? Or would this be a welcome feature addition?

Currently:
- `--skip-nx-cache` doesn't clear the entries in the cache, and we want to use the cache.
- `nx reset` only clears the local cache, not the remote cache.

**Comment by ericbf** [comment]
We are also in need of this, but for S3 cache. Anybody have any ideas for this one?

**Comment by titan-ron** [comment]
**The simplest "reset everything" trick:** bump a shared global input in `nx.json`. Every cache key is derived from the task's inputs, and global inputs feed into *every* project's hash — so changing one invalidates all keys at once and the old remote entries simply stop being looked up.

The most self-documenting way is a literal "salt" value you control. Add a custom named input and include it everywhere via `sharedGlobals`:

```jsonc
// nx.json
{
  "namedInputs": {
    "sharedGlobals": [
      "{workspaceRoot}/nx.json",
      { "runtime": "echo cache-v1" }   // bump to cache-v2, cache-v3, ... to reset
    ]
  }
}
```

Bumping `cache-v1` → `cache-v2` changes every hash to a fresh keyspace. (You can use a custom env var instead of a `runtime` value if you'd rather flip it per-CI-run without editing the file — the name is arbitrary, it just has to be listed in the inputs so it's part of the hash.)

Old entries still sit in the backend but nothing resolves to them anymore, so functionally it's a full cache reset without a remote-purge command. Pair it with a storage lifecycle/expiry policy so the orphaned entries eventually get cleaned up.
